lib/lz4.c

```diff
@@ -1434,6 +1504,26 @@ typedef enum { decode_full_block = 0, partial_decode = 1 } earlyEnd_directive;
 #undef MIN
 #define MIN(a,b)    ( (a) < (b) ? (a) : (b) )
 
+LZ4_FORCE_INLINE unsigned read_variable_length(const BYTE**ip, const BYTE* lencheck, int loop_check, int initial_check, int* error) {
```

Comment:
Could you document this function ?
Especially its arguments and their roles ?

Reply:
Documented, and changed error code to a typedef enum.